src/hera/shared/_global_config.py

```diff
@@ -9,6 +10,7 @@
 _HookMap = Dict[Type[TBase], List[Hook]]
 
 
+@dataclass
```

Comment:
What do you think about making this a Pydantic model?

Reply:
@flaviuvadan tried pydantic first, but we cannot mimic the current implementation using pydantic in a b/w compatible fashion.

Pydantic gives errors like - 

```
FAILED tests/test_hooks.py::test_container_pre_build_hooks - ValueError: "_GlobalConfig" object has no field "_pre_build_hooks"
FAILED tests/test_hooks.py::test_workflow_pre_build_hooks - ValueError: "_GlobalConfig" object has no field "_pre_build_hooks"
FAILED tests/test_workflow.py::test_hera_output[coinflip] - ValueError: "_GlobalConfig" object has no field "__dict__"
FAILED tests/test_workflow.py::test_hera_output[complex_deps] - ValueError: "_GlobalConfig" object has no field "__dict__"
FAILED tests/test_workflow.py::test_hera_output[dag_conditional_parameters] - ValueError: "_GlobalConfig" object has no field "__dict__"
FAILED tests/test_workflow.py::test_hera_output[dag_diamond_with_callable_container] - ValueError: "_GlobalConfig" object has no field "__dict__"
FAILED tests/test_workflow.py::test_hera_output[dag_diamond_with_callable_decorators] - ValueError: "_GlobalConfig" object has no field "__dict__"
```